### DISCUSSION TITLE
Shift in tap dance and simple morph

### DISCUSSION THREAD

**Discussion by anarion80** [urob/zmk-config#90] [Q&A] [answered]
Hello!

I have an issue with &dot_spc and:

```c
// tap: comma | shift + tap: semicolon | cmd + shift + tap: <
SIMPLE_MORPH(comma_morph, SFT, &kp COMMA, &comma_inner_morph)
SIMPLE_MORPH(comma_inner_morph, GUI, &kp SEMICOLON, &kp LESS_THAN)

```

in dot_spc case I don't get the key I want but the shifted version. Instead of "dot space shifted_letter" I get "> space shifted_letter".
![image](https://github.com/user-attachments/assets/9a90053f-403c-4734-844e-ea3b909ddffd)

In case of "comma, semicolon, less_than" I get `,` on tap, `:` on first tap with shift and `;` only on subsequent taps, the same with cmd+shft - the first one is always not what I want.

![image](https://github.com/user-attachments/assets/fb136610-a2b2-448a-b660-fb81016ab25a)
It's as this SHIFT is superfluous there on the first tap with it... Does anyone have any idea what I have wrong in the config?

My config: https://github.com/anarion80/zmk-config-corneish-zen/blob/main/config/base.keymap

**Comment by urob** [comment, marked as answer]
Are you on MacOS by any chance and using Karabiner? Karabiner is [known to be buggy](https://github.com/pqrs-org/Karabiner-Elements/issues/3420) in combination with mod-morph.

**Reply by anarion80** [reply]
Yes and yes.

**Reply by anarion80** [reply]
And quitting Karabiner does resolve this Shift issue...

**Reply by urob** [reply]
Yea, there is a known issue (I just added the link in the first comment). It's possible to disable Karabiner on a keyboard-basis. And in case you are only using karabiner to remap some keys, you might be able to do that [natively](https://medium.com/ryan-hanson/key-remapping-built-into-macos-c7953b1a62e4)

**Reply by anarion80** [reply]
Thanks!
